### ISSUE TITLE
[flang] include flang by default under LLVM_ENABLE_PROJECTS

### ISSUE THREAD

**Issue by h-vetinari** [llvm/llvm-project#112789]
This has been discussed numerous times, and generally was made dependent on the rename from flang-new to flang which now happened: https://github.com/llvm/llvm-project/commit/06eb10dadfaeaadc5d0d95d38bea4bfb5253e077

I've tried to find if there's an issue for this but couldn't find one (apologies if I missed something), so opening this.

**Comment by nikic** [comment]
I don't think that flang should become part of the default project list unless and until significant improvements are made to the memory usage of the flang build. A release build should not require more than 2GB of memory per thread, otherwise including it in the default build becomes a liability.

A possible alternative would be to include cmake logic that automatically reduces the parallelism of *only* the flang build to fit its unusual memory usage requirements.

**Comment by h-vetinari** [comment]
> A possible alternative would be to include cmake logic that automatically reduces the parallelism of _only_ the flang build to fit its unusual memory usage requirements.

I was under the impression that this happened already? I'm on a phone right now, but IIRC there was the introduction of a specific CMake variable for a lower default parallelism specifically for the flang build

**Comment by kiranchandramohan** [comment]
> > A possible alternative would be to include cmake logic that automatically reduces the parallelism of _only_ the flang build to fit its unusual memory usage requirements.
> 
> I was under the impression that this happened already? I'm on a phone right now, but IIRC there was the introduction of a specific CMake variable for a lower default parallelism specifically for the flang build

That patch (https://github.com/llvm/llvm-project/pull/95672) was reverted (https://github.com/llvm/llvm-project/pull/96000) since there was a failure in one of the buildbots.

**Comment by h-vetinari** [comment]
Sounds like that patch should be reattempted/relanded then? :)

**Comment by Meinersbur** [comment]
Should the default of #95672 consider the host's available memory then? The default of "0" (infinity) will catch many users by surprise and crash their system, especially since it will override `LLVM_PARALLEL_COMPILE_JOBS` which is actually derived from `LLVM_RAM_PER_COMPILE_JOB` (Since #65274).

**Comment by tstellar** [comment]
> Sounds like that patch should be reattempted/relanded then? :)

Here it is: https://github.com/llvm/llvm-project/pull/127364
